### PR TITLE
✨feat: Implement Zod Validation for Email and Password in auth/login …

### LINKED ISSUE #9008: Implement Zod Validation for Email and Password Fields in auth/login Page
### Is your proposal related to a problem?

Currently, when the email and password fields are left empty or contain invalid data in the auth/login page, the browser displays default error messages. 

Calcom-defaultBrowser

### Describe the solution you'd like

We should improve the user experience by providing custom error messages.
Implementing Zod validation for this two field
Calcom-ZodValidation

@hariombalhara

### PR DESCRIPTION
**Pull request by paahaad**
Page:- auth/login

## What does this PR do?

Fixes #9008

https://www.loom.com/share/b09b9e1ece314427b0a2e308bd3d8a64

**Environment**: Staging(main branch) / Production

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- Chore (refactoring code, technical debt, workflow improvements)
- New feature (non-breaking change which adds functionality)

@hariombalhara @PeerRich

### CODE REVIEW

File: apps/web/pages/auth/login.tsx [hunk 1]
```diff
@@ -161,14 +169,14 @@ export default function Login({
                   label={t("email_address")}
                   defaultValue={totpEmail || (router.query.email as string)}
                   placeholder="john.doe@example.com"
-                  required
+                  // Note: override the type as text because it was throwing the default browser error message, Checking email type By zod
+                  type="text"
```
Comment (hariombalhara) [inline apps/web/pages/auth/login.tsx:173]: Better way would be to keep it as is but add `noValidate` prop to Form.

File: apps/web/pages/auth/login.tsx [hunk 2]
```diff
@@ -161,14 +169,14 @@ export default function Login({
...
                   {...register("email")}
                 />
                 <div className="relative">
                   <PasswordField
                     id="password"
                     autoComplete="off"
-                    required={!totpEmail}
```
Comment (hariombalhara) [inline apps/web/pages/auth/login.tsx:171]: We can keep this as well after noValidate is added.

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Looks good except noValidate change that's pending

**Comment by parvatraj2** [comment]
Hi @hariombalhara 
Thanks for the review. I have made the required changes.

**Review by hariombalhara — APPROVED** [review]

**Pull request merged**